### DISCUSSION TITLE
Vector of functions as template arguments

### DISCUSSION THREAD

**Discussion by nuser721882** [community/community#134968] [Programming Help]
### Select Topic Area

Question

### GitHub Feature Area

Actions

### Body

Hello. 
I am working on C++ language. 
Please I have an issue while trying to store a vector of functions into template class. This class is the main class (main_class) which is linked to the main function (int main()). I create a vector of functions. For this, I have created a functor (functor_00) that returns a std::vector. Then pass this functor to an argument template class (main_class) using template<typename T, typename F> my _class = functor_00. The template main_class  has many argument like this: template<typename T, template<T, typename F>my_class =functor_00 , template<T>class vec>main_class.  The class that returns a vector of functions, which is in another files and corresponds to objective function of my problem  in turn is a functor (my_real_functor) that contains real functions of my problem. In main function I call my_real_functor like this in main function: main_class<functor_00<double, my_real_functor <double>>>. This it won't compile showing : error: type/value mismatch at argument  in template parameter list. Please how can I skip this? Please Note that I want that the vector functions of my problem would be defined in my_real_functor instead of main_class.

**Comment by queenofcorgis** [comment]
Hey there!  👋 

Thanks for posting in the GitHub Community, @nuser721882 ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by nuser721882** [comment]
can somone give a help
